test/linalg/generic.jl

```diff
@@ -310,3 +310,33 @@ end
 @test [[1, 2], [3, 4]] ≈ [[1.0-eps(), 2.0+eps()], [3.0+2eps(), 4.0-1e8eps()]]
 @test [[1, 2], [3, 4]] ≉ [[1.0-eps(), 2.0+eps()], [3.0+2eps(), 4.0-1e9eps()]]
 @test [[1,2, [3,4]], 5.0, [6im, [7.0, 8.0]]] ≈ [[1,2, [3,4]], 5.0, [6im, [7.0, 8.0]]]
+
+# Issue 22042
+# Minimal modulo number type - but not subclassing Number
+struct ModInt{n}
+        k
+        ModInt{n}(k) where {n} = new(mod(k,n))
+end
+
+Base.:+(a::ModInt{n}, b::ModInt{n}) where {n} = ModInt{n}(a.k+b.k)
+Base.:-(a::ModInt{n}, b::ModInt{n}) where {n} = ModInt{n}(a.k-b.k)
+Base.:*(a::ModInt{n}, b::ModInt{n}) where {n} = ModInt{n}(a.k*b.k)
```

Comment:
Spaces surrounding the operators might make these statements easier on wetware parsers?